### ISSUE TITLE
WebGL Rendering fails for a High BVAL Diffusion image

### ISSUE THREAD

**Issue by swederik** [cornerstonejs/cornerstone#210]
[000009.dcm.zip](https://github.com/cornerstonejs/cornerstone/files/1619549/000009.dcm.zip)

I have no idea why this is failing in WebGL, but it works fine on the canvas renderer.

I get:

````
WebGL: INVALID_OPERATION: texImage2D: type UNSIGNED_BYTE but ArrayBufferView not Uint8Array
[.Offscreen-For-WebGL-0x7fbf90cd3200]RENDER WARNING: texture bound to texture unit 0 is not renderable. It maybe non-power-of-2 and have incompatible texture filtering
````

**Comment by NicolasRannou** [comment]
Can you share the anonymized data by any chance?

I'd love to test in AMI / Papaya!
https://fnndsc.github.io/ami/#viewers_upload

Thx!

**Comment by swederik** [comment]
Sure @NicolasRannou, I think it's attached to the post above:

https://github.com/cornerstonejs/cornerstone/files/1619549/000009.dcm.zip

**Comment by seandoyle** [comment]
I've run into a similar issue with some secondary capture images that we generated - the bits allocated was 16 bit but the max pixel value was below 255.  In https://github.com/cornerstonejs/cornerstone/blob/master/src/webgl/renderer.js there is the following code: 
` if (image.maxPixelValue > 255) {
    datatype += '16';
  } else {
    datatype += '8';
  }`
So - WebGL assumes that it's an 8 bit array - but the pixels are stored in a 16 bit array. I'm not sure what the right fix would be:

1. If you know that the data only has an 8 bit range - maybe for memory's sake the data should be copied to an 8 bit array early in the data parsing pipeline.
2. Or - getImageDataType() needs to look at the DICOM bits allocated instead of max pixel size.

Not sure which is better.

**Comment by swederik** [comment]
Fixed by @Zaid-Safadi  in #253
